### ISSUE TITLE
Cannot install Docker, which is already present

### ISSUE THREAD

**Issue by marcosh** [getredash/setup#68]
I tried the installation script on my archlinux machine. I understand that such a platform is not yet supported.

Still, I get the error `This doesn't seem to be a Debian, Fedora, Ubuntu, nor RHEL (compatible) system, so this script doesn't know how to add Docker to it.`.

The point is that `docker` and `docker-compose` are already installed in the system.

It would be nice to check whether `docker` and `docker-compose` are already installed before failing with such a message

**Comment by justinclift** [comment]
Good point.  Yeah, the script is currently made for fairly newly installed systems with not much existing already on them. aka newly installed VM or similar

That being said, it's probably not all that hard to check if docker is installed before trying to install it.

Out of curiosity, are you any good at shell scripting? :smile:

**Comment by lucydodo** [comment]
Are you saying that it's better to make sure user have Docker installed before checking if it's a distribution that our script knows how to install Docker, and if so, go through the installation process regardless of the supported distribution?

**Comment by lucydodo** [comment]
@justinclift Oh, we commented at almost the same time. LOL

**Comment by lucydodo** [comment]
Hi @marcosh,
This issue will be resolved by the [9723922](https://github.com/getredash/setup/commit/9723922882e5eab5c891623e3826b32a279a4760) commit. Could you please check it out and let us know? :)

**Comment by marcosh** [comment]
Hi, thanks for the quick response!

I was able to use the script version from `master` with some minor manual tweaks:
- I had to install `pwgen` manually
- I had to set the `PROFILE` variable in the script

Apart from that, everything worked fine!

Just a random though, did you ever think about using Nix instead of Docker? That should alleviate the need to consider all the differences between the different Linux distros. I'm not a Nix expert though, so I might very well be wrong on this
